### DISCUSSION TITLE
Coding Style

### DISCUSSION THREAD

**Discussion by ElectronicsArchiver** [atom-ide-community/atom#389] [Ideas] [answered]
<br>

As this is a new beginning, I was wondering if the style
could be adjused to be 'prettier' as well, pun intended.

<br>

*Generally speaking I don't think very highly of linters*
*and the like, to me not going through your code and doing*
*at least a little bit of clean up - and finding possible*
*problems - is a bit backwards.*

<br>

### My Suggestion

Instead of writing up guidelines that no one wants to read,
how about just refering to already well formatted files.
( *Once there are some* )

***Work by example, that is.***

<br>

### Style

Don't enforce one strict style, simply 
show things that make it better.

<br>

##### Columns

Simply staying below 60 - 80

##### Editorconfig

Use editorconfig for basic consistensy

##### Writing small / clear if else return statements

```javascript
startIdForScope(scope){
    return (scope === this.scopeName)
        ? -1
        : null ;
},

startIdForScope(scope){
    return (scope === this.scopeName)
        ? -1 : null ;
},  
```

##### Give stuff space

Vertical space is cheap / easy to traverse~

```javascript
const path = require('path');
const fs = require('fs-plus');


const { log } = console;


const hasWriteAccess = (directory) => {
    
    const testPath = path
        .join(directory,'write.test');
  
    const date = new Date()
        .toISOString();
  
    try {
        fs.writeFileSync(testPath,date,{ flag: 'w+' });
        fs.unlinkSync(testPath);
    } catch (error) { return false; }
    
    return true;
};
```

<br>

## Guide

Creating a nice, simply & readable style guide
would be a good way to convey that, I'd say.

*Unlike https://github.com/atom/atom/blob/master/CONTRIBUTING.md for example*

<br>

**Comment by aminya** [comment, marked as answer]
We already have a coding style, and it is Prettier. Run `script/lint --fix`. 

This is our prettier config used in all the packages (except for Atom itself). We don't want to change the prettier config used in Atom because that will create conflicts in the pull requests.
https://github.com/atom-community/prettier-config-atomic

**Reply by ElectronicsArchiver** [reply]
And what about Atom itself?

**Reply by aminya** [reply]
Atom uses prettier through eslint (the default settings)
https://github.com/atom-community/atom/blob/7eaebe0eea3c703ac8041931ac663864cbe2d090/.eslintrc.json#L28
